core/src/avm2.rs

```diff
@@ -256,16 +256,12 @@ impl<'gc> Avm2<'gc> {
         context: &mut UpdateContext<'_, 'gc>,
         mut f: impl FnMut(DisplayObject<'gc>, &mut UpdateContext<'_, 'gc>),
     ) {
-        let mut i = 0;
         // FIXME - should we handle movies added while we're looping?
-        let total = context.avm2.orphan_objects.len();
-
         // We cannot use an iterator, as it would conflict with the mutable borrow of `context`.
-        while i < total {
-            if let Some(dobj) = valid_orphan(context.avm2.orphan_objects[i], context.gc_context) {
+        for orphan in context.avm2.orphan_objects.clone() {
```

Comment:
:(

Reply:
This was a quick fix for a crash - I think I can introduce an `Rc` and use `Rc::make_mut`, like we already do for render list iteration

Reply:
I split out this change into https://github.com/ruffle-rs/ruffle/pull/11414